inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -35,8 +35,8 @@ def application_add(args: argparse.Namespace) -> str:
 
     manifest += '<repo><repos>'
 
-    for source in args.sources:
-        manifest += '<source_pkg>' + source + '</source_pkg>'
+    source_tags = [f'<source_pkg>{source}</source_pkg>' for source in args.sources]
+    manifest += ''.join(source_tags)
 
     manifest += ('</repos>'
                  f'{create_xml_tag(arguments, "filename")}</repo>'
```

Comment:
**Suggestion:** Consider checking if 'args.filename' is not None or empty before adding it to the XML manifest to prevent creating an invalid XML element. [possible issue]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    manifest += '</repos>'
    if args.filename:
        manifest += f'{create_xml_tag(arguments, "filename")}'
    manifest += '</repo>'
```